src/main/io/gps.c

```diff
@@ -688,79 +1085,156 @@ void gpsInitUblox(void)
 
             if (gpsData.ackState == UBLOX_ACK_IDLE) {
                 switch (gpsData.state_position) {
+                    case UBLOX_DETECT_UNIT:
+                        if (gpsData.unitVersion == UBX_VERSION_UNDEF) {
+                            ubloxSendClassMessage(CLASS_MON, MSG_MON_VER, 0);
+                            break;
+                        }
+                        gpsData.state_position++;
+                        break;
                     case UBLOX_INITIALIZE:
-                        gpsData.ubloxUsePVT = true;
-                        gpsData.ubloxUseSAT = true;
+                        gpsData.updateRate = gpsConfig()->gps_update_rate_hz;
+
+                        ubloxSetNavRate(1, 1, 1);
+
+                        gpsData.ubloxUsePVT = gpsData.unitVersion > UBX_VERSION_M6;
+                        gpsData.ubloxUseSAT = gpsData.unitVersion > UBX_VERSION_M8;
```

Comment:
Even if `UsePVT` and `UseSAT` are correct, the does not read correctly, which says

**if unit version is greater than M6, then use PVT**
**if unit version is greater than M8, then use SAT**

Which seems incorrect. 

Both PVT and SAT are part of [u-blox8-M8](https://content.u-blox.com/sites/default/files/products/documents/u-blox8-M8_ReceiverDescrProtSpec_UBX-13003221.pdf), and supported on:
• u-blox 8 / u-blox M8 protocol versions 15, 15.01, 16, 17, 18, 19, 19.1, 19.2, 20, 20.01, 20.1, 20.2, 20.3, 22, 22.01, 23 and 23.01


[u-blox 6 does not support either](https://content.u-blox.com/sites/default/files/products/documents/u-blox6_ReceiverDescrProtSpec_%28GPS.G6-SW-10018%29_Public.pdf)


Could this piece please fixed?

Reply:
UBX-NAV-PVT was introduced in protocol 14.00, which corresponds to the M7 platform release.
The condition `gpsData.ubloxUsePVT = gpsData.unitVersion > UBX_VERSION_M6` is correct for this.

UBX-NAV_SAT was introduced in protocol 15.00, which corresponds to the M8 platform release.
The condition `gpsData.ubloxUseSAT = gpsData.unitVersion > UBX_VERSION_M8` is not correct for this.
It should be `gpsData.ubloxUseSAT = gpsData.unitVersion > UBX_VERSION_M7`.

Reply:
Thanks for the clarification, @ZzyzxTek, you are right, `PVT` was introduced at [M7 indeed](https://content.u-blox.com/sites/default/files/products/documents/u-blox7-V14_ReceiverDescriptionProtocolSpec_%28GPS.G7-SW-12001%29_Public.pdf).

On another note, I think being clear on intentions in code is better, so if we know it works on 8 and above, then we should say `gpsData.unitVersion >= UBX_VERSION_M8` rather than `gpsData.unitVersion > UBX_VERSION_M7`

Translating it into words, it is `use it on version 8 or above` instead of `use it on versions later than 7` (which theoretically could be 7.5 or whatever compares greater than 7)

Reply:
fixed in upcoming changes.

Reply:
Please re-check, should be correct now